dashboard/streamlit_app.py

```diff
@@ -1,73 +1,177 @@
+    """
+    st.title("Plant Temperature Bar Chart")
+    st.write("Bar chart showing temperature values for different plants.")
 
-    return archived_plants_data
+    plt.figure(figsize=(10, 6))
+    sns.barplot(x="species", y="temperature", data=data_df)
+    plt.xlabel("Species")
+    plt.ylabel("Temperature")
+    plt.xticks(rotation=45)
+    plt.title("Temperature Values for Different Plants")
+    st.pyplot(plt)
 
 
-def sidebar(db_connection: connection) -> None:
-    """Function that sets up the sidebar for the dashboard"""
-    st.sidebar.title("ONTHEB-RINK-OFEXTINCTION")
-    st.sidebar.markdown("An app for Liverpool Natural History Museum,\
-                        monitoring the health of our plants")
+def plot_soil_moisture_for_plants(data_df) -> None:
+    """
+    creates a bar chart with moisture in y axis
+    and the plant name in x axis
+    """
+    st.title("Plant Moisture Bar Chart")
+    st.write("Bar chart showing Moisture values for different plants.")
 
-    data = get_live_data(db_connection)
+    plt.figure(figsize=(10, 6))
+    sns.barplot(x="species", y="soil_moisture", data=data_df)
+    plt.xlabel("Species")
+    plt.ylabel("Soil Moisture")
+    plt.xticks(rotation=45)
+    plt.title("Soil Moisture Readings for Different Plants")
+    st.pyplot(plt)
 
-    toggle_on = st.sidebar.toggle(label="Toggle for Archived Data", value=False, )
-    if toggle_on:
-        data = get_archived_data()
 
-    with st.sidebar.expander("Expand to see information about the plants"):
-        st.markdown("Lorem Ipsum")
-    return data
+def handle_sidebar_options(plant_data_df) -> None:
+    """Displays initial 10 rows of the data with options for next 10 and previous 10."""
+    if "start_index" not in st.session_state:
+        st.session_state.start_index = 0
 
+    rows_to_show = 10
 
-if __name__ == "__main__":
+    with st.sidebar:
+        st.sidebar.title("Pagination")
+        if st.button("Show Previous 10 Rows"):
+            st.session_state.start_index = max(
+                0, st.session_state.start_index - rows_to_show)
 
-    load_dotenv()
-    config = environ
+        if st.button("Show Next 10 Rows"):
+            st.session_state.start_index = min(st.session_state.start_index + rows_to_show,
+                                               len(plant_data_df) - rows_to_show)
 
-    conn = connect(
-            user=config["DATABASE_USERNAME"],
-            password=config["DATABASE_PASSWORD"],
-            host=config["DATABASE_IP"],
-            port=config["DATABASE_PORT"],
-            database=config["DATABASE_NAME"],
-        )
+    data_to_show = plant_data_df[st.session_state.start_index:
+                                 st.session_state.start_index + rows_to_show]
+    plot_temp_for_plants(data_to_show)
+    plot_soil_moisture_for_plants(data_to_show)
 
-    plant_data = sidebar(conn)
 
-    st.markdown(plant_data.style.hide(axis="index").to_html(), unsafe_allow_html=True)
+if __name__ == "__main__":
```

Comment:
If you add `st.set_page_config(page_title="Plant Monitoring Dashboard", layout="wide")` this will provide a title that appears in the tab in the web browser.